src/components/Api.js

```diff
@@ -0,0 +1,193 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+        this._baseUrl = baseUrl;
+        this._headers = headers;
+    }
+    //profile and avatar 
+    async getProfile() {
+        return fetch(this._baseUrl + "/users/me/", 
+        {
+            method: "GET",
+            headers: {
+                authorization: this._headers.authorization,
+            },
+        })
+        .then((res) => {
+            if (res.ok) {
+                return res.json();
+            }
+            console.log("Error");
+            return Promise.reject(`Error : ${res.status}`);
+        }) 
+        .then((result) => {
+            console.log(result);
+            return result;
+        })
+        .catch((err) => console.error(err));
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.